src/Compilers/Test/Utilities/CSharp/CSharpTestSource.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Linq;
+
+namespace Microsoft.CodeAnalysis.CSharp.Test.Utilities
+{
+    /// <summary>
+    /// Represents the source code used for a C# test. Allows us to have single helpers that enable all the different ways
+    /// we typically provide source in testing.
+    /// </summary>
+    public readonly struct CSharpTestSource
+    {
+        public static CSharpTestSource None => new CSharpTestSource(null);
+
+        public object Value { get; }
+
+        private CSharpTestSource(object value)
+        {
+            Value = value;
+        }
+
+        public SyntaxTree[] GetSynatxTrees(CSharpParseOptions parseOptions, string sourceFileName = "")
```

Comment:
Typo: `GetSynatxTrees`

Reply:
That was just a trick to see if you were paying attention. Really. Not Joking ... yeah we all know what happened here.